internal/resources/iampolicy/scope.go

```diff
@@ -0,0 +1,197 @@
+			}
+		}
+	}
+
+	if v, ok := scopeData[namespaceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopedFullnameData = &scopedFullname{
+				scope:             namespaceScope,
+				fullnameNamespace: constructNamespaceFullname(v1),
+			}
+		}
+	}
+
+	return scopedFullnameData
+}
+
+func flattenScope(scopedFullname *scopedFullname) (data []interface{}) {
+	if scopedFullname == nil {
+		return data
+	}
+
+	flattenScopeData := make(map[string]interface{})
+
+	switch scopedFullname.scope {
+	case organizationScope:
+		flattenScopeData[organizationKey] = flattenOrganizationFullname(scopedFullname.fullnameOrganization)
+	case clusterGroupScope:
+		flattenScopeData[clusterGroupKey] = flattenClusterGroupFullname(scopedFullname.fullnameClusterGroup)
+	case clusterScope:
+		flattenScopeData[clusterKey] = flattenClusterFullname(scopedFullname.fullnameCluster)
+	case workspaceScope:
+		flattenScopeData[workspaceKey] = flattenWorkspaceFullname(scopedFullname.fullnameWorkspace)
+	case namespaceScope:
+		flattenScopeData[namespaceKey] = flattenNamespaceFullname(scopedFullname.fullnameNamespace)
+	case unknownScope:
+		log.Fatalf("[ERROR]: No valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scopesAllowed[:], `, `))
+	}
+
+	return []interface{}{flattenScopeData}
+}
+
+func validateScope(_ context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(scopeKey)
+	if !ok {
+		return fmt.Errorf("scope: %v is not valid: minimum one valid scope block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("scope data: %v is not valid: minimum one valid scope block is required", data)
+	}
+
+	scopeData := data[0].(map[string]interface{})
+	scopesFound := make([]string, 0)
+	scopesAllowed := []string{organizationKey, clusterGroupKey, clusterKey, workspaceKey, namespaceKey}
+
+	if v, ok := scopeData[organizationKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, organizationKey)
+		}
+	}
+
+	if v, ok := scopeData[clusterGroupKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterGroupKey)
+		}
+	}
+
+	if v, ok := scopeData[clusterKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterKey)
+		}
+	}
+
+	if v, ok := scopeData[workspaceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, workspaceKey)
+		}
+	}
+
+	if v, ok := scopeData[namespaceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, namespaceKey)
+		}
+	}
+
+	if len(scopesFound) == 0 {
+		return fmt.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v", strings.Join(scopesAllowed[:], `, `))
```

Comment:
Lint is failing because the array is of fixed length; use the one declared with the schema to avoid this.